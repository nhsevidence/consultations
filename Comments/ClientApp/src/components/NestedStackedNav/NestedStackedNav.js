import React, { PureComponent } from "react";
import { Link } from "react-router-dom";

export class NestedStackedNav extends PureComponent {

	renderNavRow = (item) => {
		if (item.children) {
			return (
				<ListItem key={item.to}>
					<ListLink item={item}/>
					<ListWrapper>
						{item.children.map((this.renderNavRow))}
					</ListWrapper>
				</ListItem>
			);
		} else {
			return (
				<ListItem key={item.to}>
					<ListLink item={item}/>
				</ListItem>
			);
		}
	};

	render() {
		const navigationStructure = this.props.navigationStructure;
		if (!navigationStructure) {
			return null;
		}
		return (
			<nav className="stacked-nav NestedStackedNav">
				<ListWrapper>
					{navigationStructure.map(this.renderNavRow)}
				</ListWrapper>
			</nav>
		);
	}

}

const ListLink = (props) => (
	<Link aria-current={props.item.current ? "page" : null} to={props.item.to}>
		<span data-g="11">{props.item.title}</span>
		{props.item.marker &&
			<span className="text-right" data-g="1">({props.item.marker})</span>
		}
	</Link>
);

const ListWrapper = (props) => (
	<ul className="stacked-nav__list">{props.children}</ul>
);

const ListItem = (props) => (
	<li className="grid grid--gutterless stacked-nav__list-item">{props.children}</li>
);
