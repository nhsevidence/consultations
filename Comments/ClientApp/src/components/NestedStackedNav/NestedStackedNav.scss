.NestedStackedNav {
  li > ul {
    margin-left: rem($spacing-medium);
  }
}
