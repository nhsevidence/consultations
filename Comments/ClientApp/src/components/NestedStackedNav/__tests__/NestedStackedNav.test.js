import React from "react";
import {MemoryRouter} from "react-router-dom";
import {mount} from "enzyme";
import toJson from "enzyme-to-json";
import {NestedStackedNav} from "../NestedStackedNav";

describe("[ClientApp] ", () => {
	describe("NestedStackedNav ", () => {
		const props = {
			navigationStructure:
				[
					{
						title: "consultation name",
						to: "/admin/questions/1/0",
						marker: 1,
						current: true,
						children: [
							{
								title: "consultation doc 1",
								to: "/admin/questions/1/1",
								marker: 1,
								current: false,
							},
							{
								title: "consultation doc 2",
								to: "/admin/questions/1/2",
								marker: 1,
								current: false,
							},
							{
								title: "consultation doc 3",
								to: "/admin/questions/1/3",
								marker: 1,
								current: false,
							},
							{
								title: "consultation doc 4",
								to: "/admin/questions/1/4",
								marker: 1,
								current: false,
							},
						],
					},
				],
		};

		it("should match the snapshot", () => {
			const wrapper = mount(
				<MemoryRouter>
					<NestedStackedNav {...props} />
				</MemoryRouter>,
			);
			expect(
				toJson(wrapper, {
					noKey: true,
					mode: "deep",
				})
			).toMatchSnapshot();
		});

		it("should render no markup if no documents are supplied", function () {
			const props = {};
			const wrapper = mount(
				<MemoryRouter>
					<NestedStackedNav {...props} />
				</MemoryRouter>,
			);
			const el = wrapper.find(".NestedStackedNav");
			expect(el.length).toEqual(0);
		});
	});
});
