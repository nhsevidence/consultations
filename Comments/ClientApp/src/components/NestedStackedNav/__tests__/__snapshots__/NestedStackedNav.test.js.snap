// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  NestedStackedNav  should match the snapshot 1`] = `
<nav
  className="stacked-nav NestedStackedNav"
>
  <ul
    className="stacked-nav__list"
  >
    <li
      className="grid grid--gutterless stacked-nav__list-item"
    >
      <a
        aria-current="page"
        href="/admin/questions/1/0"
        onClick={[Function]}
      >
        <span
          data-g="11"
        >
          consultation name
        </span>
        <span
          className="text-right"
          data-g="1"
        >
          (
          1
          )
        </span>
      </a>
      <ul
        className="stacked-nav__list"
      >
        <li
          className="grid grid--gutterless stacked-nav__list-item"
        >
          <a
            aria-current={null}
            href="/admin/questions/1/1"
            onClick={[Function]}
          >
            <span
              data-g="11"
            >
              consultation doc 1
            </span>
            <span
              className="text-right"
              data-g="1"
            >
              (
              1
              )
            </span>
          </a>
        </li>
        <li
          className="grid grid--gutterless stacked-nav__list-item"
        >
          <a
            aria-current={null}
            href="/admin/questions/1/2"
            onClick={[Function]}
          >
            <span
              data-g="11"
            >
              consultation doc 2
            </span>
            <span
              className="text-right"
              data-g="1"
            >
              (
              1
              )
            </span>
          </a>
        </li>
        <li
          className="grid grid--gutterless stacked-nav__list-item"
        >
          <a
            aria-current={null}
            href="/admin/questions/1/3"
            onClick={[Function]}
          >
            <span
              data-g="11"
            >
              consultation doc 3
            </span>
            <span
              className="text-right"
              data-g="1"
            >
              (
              1
              )
            </span>
          </a>
        </li>
        <li
          className="grid grid--gutterless stacked-nav__list-item"
        >
          <a
            aria-current={null}
            href="/admin/questions/1/4"
            onClick={[Function]}
          >
            <span
              data-g="11"
            >
              consultation doc 4
            </span>
            <span
              className="text-right"
              data-g="1"
            >
              (
              1
              )
            </span>
          </a>
        </li>
      </ul>
    </li>
  </ul>
</nav>
`;
